custom_components/tesla_custom/base.py

```diff
@@ -17,22 +17,23 @@ class TeslaBaseEntity(CoordinatorEntity):
 
     _attr_attribution = ATTRIBUTION
     _attr_has_entity_name = True
+    _enabled_by_default: bool = True
+    type: str
 
     def __init__(
         self, hass: HomeAssistant, coordinator: TeslaDataUpdateCoordinator
     ) -> None:
         """Initialise the Tesla device."""
         super().__init__(coordinator)
         self._coordinator: TeslaDataUpdateCoordinator = coordinator
-        self._enabled_by_default: bool = True
         self.hass = hass
-        self.type = None
         self._memorized_unique_id = None
```

Comment:
This could likely be cleaned up more since `CoordinatorEntity` takes care of setting `self.coordinator` and `self.hass` is set in `async_added_to_hass` but that would require more invasive refactoring and I didn't want to make this larger

Reply:
No worries. Thanks for the cleanup you did do.

Reply:
Thanks. Did this in https://github.com/alandtse/tesla/pull/671

Also finally had some time to look into why startup was so slow when the car is asleep https://github.com/zabuldon/teslajsonpy/pull/416